lms/djangoapps/grades/course_grade_factory.py

```diff
@@ -193,26 +197,27 @@ def _update(user, course_data, force_update_subsections=False):
                 passed=course_grade.passed,
             )
 
-        COURSE_GRADE_CHANGED.send_robust(
-            sender=None,
-            user=user,
-            course_grade=course_grade,
-            course_key=course_data.course_key,
-            deadline=course_data.course.end,
-        )
-        if course_grade.passed:
-            COURSE_GRADE_NOW_PASSED.send(
-                sender=CourseGradeFactory,
-                user=user,
-                course_id=course_data.course_key,
-            )
-        else:
-            COURSE_GRADE_NOW_FAILED.send(
-                sender=CourseGradeFactory,
+        if send_grade_signals:
+            COURSE_GRADE_CHANGED.send_robust(
+                sender=None,
                 user=user,
-                course_id=course_data.course_key,
-                grade=course_grade,
+                course_grade=course_grade,
+                course_key=course_data.course_key,
+                deadline=course_data.course.end,
             )
+            if course_grade.passed:
+                COURSE_GRADE_NOW_PASSED.send(
+                    sender=CourseGradeFactory,
+                    user=user,
+                    course_id=course_data.course_key,
```

Comment:
why is using `send_grade_signals=False` better than sending the course_grade in the signal? and removing `course_grade = _get_course_grade(user, course_key, send_grade_signals=False)`? :thinking:

Reply:
Nice question! 
I didn’t try it, but now I want to 🙂
I will give it a try ASAP

Reply:
I've tried this and can say that:
- this works perfectly for certificates generated automatically
- this will not work for certificates generated manually (this case requires to call `_get_course_grade` 'cause there is no other way to get grades)

Reply:
And calling it will lead to sending the signal and checking the certificate status again

Reply:
Thanks for testing that out!